### DISCUSSION TITLE
Withdrawl and setting destination account to liability

### DISCUSSION THREAD

**Discussion by bartekd123** [firefly-iii/firefly-iii#4710] [General]
Hi James,

A small change that I noticed in the latest build (or one of the last couple). Here is what I am doing:

- Created liability account called Mortgage
- When the mortgage is paid, out of an asset account, there is a rule that changes the destination account to Mortgage

Previous Version: I can see that the source is the asset account, and destination is the Mortgage (liability), and I can see the amount of mortgage is decreasing (eg graph and total amount)
Newly updated version: Source account is good, but the destination account is creating a new expense account called "mortgage". This is not linking to the liability account, as it isn't seeing it

When I was setting this up previously, I couldn't set it to a transfer (maybe that is the right way of doing it?) so did it this way, and it has been working all this time. 

I have a couple liabilities, and they all behave the same way. Wondering if my old way is not the right way of going about it, or if this is a bug in the latest version?

**Comment by JC5** [comment]
No that should work as you describe it. I'll do some digging. It could be a bug.

**Reply by JC5** [reply]
Yes, that's a bug. Will be part of the next release :+1:

**Reply by bartekd123** [reply]
Hi James. I updated to the latest and it is still not working, its still not linking to my liability.

Here is more details if you need it:
Using CSV Importer to import the following transaction:

Date,Transaction,Name,Memo,Amount
4/9/2021,OTHER,EFT Withdrawal to bank,To bank,-123.45

Rule:
 - Trigger - Description is "EFT Withdrawal to bank"
 - Rule will - Link to Bill, Set category, set destination account to "Mortgage" (Mortgage is my Liability)

**Reply by bartekd123** [reply]
Actually, nevermind. I deleted the old transactions, and when re-importing it was linking back to the "expense" account that it created last time. I deleted the transactions and the account and re-imported. Looks good now. Thanks!
